modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -246,6 +246,11 @@
                    :semantic-version
                    (driver.u/semantic-version-gte [5]))))
 
+(defmethod driver/database-supports? [:mongo :datetime-diff] [_ _ db]
+  (boolean (some-> (:dbms_version db)
+                   :semantic-version
+                   (driver.u/semantic-version-gte [5]))))
```

Comment:
Nil punning has been removed on the v3 removal branch.

Reply:
Thanks, updated here https://github.com/metabase/metabase/pull/27042/commits/7175935adf224dc9410b6e059fae5ad8dde82492